knowledge_giver.rb

```diff
@@ -0,0 +1,50 @@
+require_relative 'class_helpers'
+
+class KnowledgeGiver < Person
+  attr_reader  :salary, :target_raise
+  DEFAULT_PHASE = 3
+
+  def initialize(options={})
+    super
+    @target_raise = options.fetch(:target_raise, self.class::DEFAULT_TARGET)
```

Comment:
is this value being passed in the options hash or set in the class? You are setting that value on line 5 so why would you be looking for that value in the optional params?